docs/get-started/upgrading.md

```diff
@@ -49,8 +49,7 @@ async def do_search():
         date_filter = filters.date_range_filter('acquired', gte=datetime.fromisoformat("2022-11-18"), lte=datetime.fromisoformat("2022-11-21"))
         cloud_filter = filters.range_filter('cloud_cover', lte=0.1)
         download_filter = filters.permission_filter()
-        search_results = await client.search(["PSScene"], filters.and_filter([date_filter, cloud_filter, download_filter]))
-    return [item async for item in search_results]
+    return [item async for item in client.search(["PSScene"], filters.and_filter([date_filter, cloud_filter, download_filter]))]
```

Comment:
@kevinlacaille you've got a valid point. The state of the art for usage is 

```python
async for item in client.search(...):
    # do stuff
```

And this can also be turned into a list comprehension like `[item async for item in client.search(...)]`.

The code on line 52 could be made easier to read if the filter construction were lifted out, maybe? Like

```python
search_filters = filters.and_filter([date_filter, cloud_filter, download_filter])
async for item in client.search(["PSScene"], search_filters):
    # do stuff
```

Reply:
This certainly isn't worth re-opening this ticket for, but I do agree what @sgillies wrote would be easier to read and maintain in the future 👍